Kiosk/Bid Fulfillment/LoadingViewController.swift

```diff
@@ -55,23 +60,47 @@ class LoadingViewController: UIViewController {
 
         // The view model will perform actions like registering a user if necessary,
         // placing a bid if requested, and polling for results.
-        viewModel.performActions().finally { [weak self] in
-            // Regardless of error or completion. hide the spinner.
-            self?.spinner.hidden = true
-        }.subscribeError({ [weak self] (error) -> Void in
-            logger.log("Bidder error \(error)")
-            self?.bidderError(error)
-        }, completed: { [weak self] () -> Void in
-            logger.log("Bid placement and polling completed")
-            self?.finishUp()
-        })
+        viewModel.performActions().subscribe(onNext: nil,
+            onError: { [weak self] error in
+                logger.log("Bidder error \(error)")
+                self?.bidderError(error as NSError)
+            },
+            onCompleted: { [weak self] in
+                logger.log("Bid placement and polling completed")
+                self?.finishUp()
+            },
+            onDisposed: { [weak self] in
+                // Regardless of error or completion. hide the spinner.
+                self?.spinner.hidden = true
+            })
+            .addDisposableTo(rx_disposeBag)
+    }
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        _viewWillDisappear.onNext()
```

Comment:
It's actually sending `Void()`, which is just an empty tuple `()`, so Swift infers from the empty parameter list that that's what I want to send (only for `Void` parameter types).